baby-liminal-extension/src/backend/mod.rs

```diff
@@ -78,12 +101,38 @@ where
     }
 
     /// Handle `verify` chain extension call.
-    pub fn verify<BackendExecutor: BackendExecutorT, Environment: EnvironmentT>(
+    pub fn verify<
+        BackendExecutor: BackendExecutorT,
+        Environment: EnvironmentT,
+        Weighting: WeightInfo,
+    >(
         mut env: Environment,
     ) -> ChainExtensionResult<RetVal> {
-        // todo: charge weight, validate args
+        // ------- Pre-charge optimistic weight. ---------------------------------------------------
+        let pre_charge = env.charge_weight(Weighting::verify())?;
```

Comment:
`Weighting::verify()` should take some arguments, not sure which at this point, but the cost depends on something like "length of public inputs". Maybe not so crucial for the PoC though, but important to correct later.

Reply:
sure, it will, but we are still missing benchmarks and thus it's no-arg